explanation/journey-of-a-change-to-production.rst

```diff
@@ -0,0 +1,317 @@
+
+What is buildbot?
+-----------------
+
+`Buildbot <https://buildbot.net>`_ is a continuous integration system that we
+use for running the Launchpad test suite before promoting the changes in the
+``master``, ``db-devel`` branches to the ``stable``, ``db-stable`` branches
+(changes to production are deployed from these branches) respectively.
+
+Our deployment of buildbot has 2 or more build configurations defined and these
+configurations have names like ``lp-devel-focal`` and ``lp-db-devel-focal``,
+where the former runs the test suite on the ``master`` branch (code changes)
+and the latter runs the test suite on the ``db-devel`` branch (database
+changes).
+
+There is a `buildbot poller script`_ that runs on the buildbot buildmaster and
+merges new commits that have completed a successful Launchpad test suite run
+into the appropriate stable branch.
+
+.. _buildbot poller script: https://bazaar.launchpad.net/~canonical-launchpad-branches/lpbuildbot/public/view/head:/buildbot-poll.py
+
+What is deployable?
+-------------------
+
+`Deployable <https://launchpad.net/isitdeployable>`_ is the web application that
+we use to track each revision of code, whether it is ready to be deployed, and
+the details of the commits included in each deployment. There is a
+`Launchpad deployable project`_ (tracking the ``stable`` branch) and a
+`Launchpad DB deployable project`_ (tracking the ``db-stable`` branch.)
+
+.. _Launchpad deployable project: https://deployable.ols.canonical.com/project/launchpad
+.. _Launchpad DB deployable project: https://deployable.ols.canonical.com/project/launchpad-db
+
+The deployable page lists the commits in the corresponding stable branch that
+haven't been deployed to the production environment. To deploy a specific
+commit, that commit and all its ancestors that are listed in that page should
+be tested by a Launchpad engineer (usually the author of the commit) in the
+``staging`` or ``qastaging`` environments and marked as deployable using the
+button under each commit.
+
+Then during the deployment process, a deployment request is created on this
+web application to track the commits that are included in that deployment. And
+after the deployment is completed, the deployment request is marked as deployed.
+
+The deployment requests on this application are solely used for tracking and do
+not affect the actual deployment process in any way.
+
+What are the Launchpad pre-production environments?
+---------------------------------------------------
+
+At the time of writing, Launchpad has 2 pre-production environments,
+`staging`_ and `qastaging`_. The ``staging`` environment is primarily used to
+test the database changes whereas the ``qastaging`` environment is used to
+test the code changes. These are deployed on a Canonical ProdStack environment
+and the Launchpad team members have shell access to these environments via the
+VPN.
+
+We have cron jobs in the Launchpad bastion environment to automatically deploy
```

Comment:
Could you please name the scripts, the exact location, and links to the scripts source code?